code/json.sls

```diff
@@ -30,172 +30,185 @@
 
 (library (json)
   (export json-write
-	  json-read)
+          json-read)
   
   (import
-    (except (rnrs) define-record-type error)
-    (rnrs r5rs)
-    (rnrs mutable-pairs)
-    (only (srfi :1 lists) lset-union append-map fold)
-    (srfi :6 basic-string-ports)
-    (srfi :9 records)
-    (srfi :23 error)
-    (only (chezscheme) include)
-    (packrat))
+   (except (rnrs) define-record-type error)
+   (rnrs r5rs)
+   (rnrs mutable-pairs)
+   (only (srfi :1 lists) lset-union append-map fold iota)
+   (srfi :6 basic-string-ports)
+   (srfi :9 records)
+   (srfi :23 error)
+   (only (chezscheme) include)
+   (packrat))
 
   (define json-write
     (let ()
+      (define indent-level 0)
+      (define (display-level)
+        (display
+         (apply string-append (map (lambda (_) "  ") (iota indent-level)))))
```

Comment:
You'll want to pass `p` as an argument to `display-level`, as the other helper functios do. Also you definitely don't want build a string with `string-append` and `iota`. Looping with `do` `indent-level` times and calling `(put-char #\space p)` (or something along those lines) would be faster

Reply:
You could also look in to `make-string`

Reply:
Will do.

Reply:
Just looked at make string.  It is great, and exactly what we need here.